metaflow/plugins/kfp/kfp.py

```diff
@@ -337,12 +341,15 @@ def kfp_pipeline_from_flow(datastore_root: str = DATASTORE_SYSROOT_S3):
             visited = {}
 
             def build_kfp_dag(node: DAGNode, context: str, index=None):
-                kfp_component = step_to_kfp_component_map[node.name]
+                kfp_component, step_name, task_id = step_to_kfp_component_map[node.name]
```

Comment:
This is cleaner if it returns a `kfp_component` instance rather than pattern matching out the class variables.
The map is to a https://github.com/zillow/metaflow/blob/1b861af0b9c6f81522b6e53ebf6ca88860ef5f1f/metaflow/plugins/kfp/kfp.py#L18